core/server/api/configuration.js

```diff
@@ -27,7 +28,7 @@ function getBaseConfig() {
         fileStorage:    config.get('fileStorage') !== false,
         useGravatar:    !config.isPrivacyDisabled('useGravatar'),
         publicAPI:      config.get('publicAPI') === true,
-        blogUrl:        config.get('url').replace(/\/$/, ''),
```

Comment:
The problem with this is the following:
`getBaseUrl` was designed to force SSL if the users request is secure, but the config url is configured to http. In this case we force SSL and replace the protocol. So the secure option should always depend on the request. 
See https://github.com/TryGhost/Ghost/issues/6270

It would be great to add a comment to this function. 
I went again through the usages and most of the use cases only check if the config is available at all and then do an action. So nothing TODO here except of adding a comment for the `getBaseUrl` fn.